basic_module/distribution/src/core.rs

```diff
@@ -41,6 +27,12 @@ pub trait TransactionHandler {
     fn create_distribution_transaction(&self) -> Transaction;
 
     fn create_open_block_transaction(&self, header: &Header) -> Transaction;
+}
+
+pub trait BlockExecutor {
+    fn open_block(&self, context: &mut dyn SubStorageAccess, header: &Header, verified_crime: &[VerifiedCrime]);
+
+    fn execute_transactions(&self, context: &mut dyn SubStorageAccess, transactions: &[Transaction]);
```

Comment:
@HoOngEe @somniumism What was our conclusion on managing context?
Should `context` explicitly exposed to each module?

Reply:
Do you mean the explicit `context` parameter? I think it is desirable though currently not supported.